lib/std/json/dynamic_test.zig

```diff
@@ -288,3 +289,42 @@ test "polymorphic parsing" {
     try testing.expect(tree.div.color == .blue);
     try testing.expectEqualStrings("Cancel", tree.div.children[1].button.caption);
 }
+
+test "long object value" {
+    const value = "01234567890123456789";
+    const doc = "{\"key\":\"" ++ value ++ "\"}";
+    var fbs = std.io.fixedBufferStream(doc);
+    var reader = smallBufferJsonReader(testing.allocator, fbs.reader());
+    defer reader.deinit();
+    var parsed = try parseFromTokenSource(Value, testing.allocator, &reader, .{});
+    defer parsed.deinit();
+
+    try testing.expectEqualStrings(value, parsed.value.object.get("key").?.string);
```

Comment:
Side question: is it generally considered acceptable to rely on safety checks in tests (such as the `.?` here)? I followed the example of the other tests in this file, but this does prevent the test from working correctly in `ReleaseFast` or `ReleaseSmall` mode and results in somewhat less optimal test failure output.

Reply:
i understand how it's kinda confusing in this case, since you could test the `get` as well, but i can't imagine a plausible scenario where the `get` would cleanly return null given this testing code. the likely bugs that would cause this test to fail would either trip the `expect` with corrupted data or completely crash, so i don't think there's any value in deliberately testing the `.?`.

it's expected that in some cases tests will erroneously pass in release fast mode, because undefined behavior often manifests as silently working, and that's ok. for std lib tests like this, all modes have to pass or else the ci checks don't allow a pr to merge, so we're all good here.

Reply:
OK, that makes sense, thanks for clarifying! I appreciate the insight.